### DISCUSSION TITLE
Disabling mod with a small string, something whispered to player?

### DISCUSSION THREAD

**Discussion by Lost-Feight** [AdvancedXRay/XRay-Mod#181] [Ideas]
Hi, basically as title, xray mods are the bane of public servers, is there any way you can make a disabling toggle serverside while they are on a server?

like if the server sends a whisper with xyzabc or a strange combination &%399d then it disables the mod until they rejoin a new server / singleplayer world?

**Comment by MichaelHillcox** [comment]
The problem with this kinda idea is that the mod is open source, if I add protections in for servers, another person with simply clone the project, remove the protections and then rebuild the mod. It's very hard to actually prevent xray on servers. You can make a plugin that can send a fake packet to the client upon connection that acts like the forge mod validation packet, you can then get a list of mods from the client and kick / ban if this mod is in the list.

Of course, this falls into the same problem, clone -> change mod id -> build.

For users without the ability to do this, I can likely prevent the mod running on servers but it would hurt anyone using it for legit reasons on a server like checking world gen for pack devs, etc.

**Reply by Lost-Feight** [reply]
ok, yeah i get it, the more you push the more they push and nothing really changes except more work for no reason... got it, thanks for the reply

**Reply by MichaelHillcox** [reply]
Yeah, it's a problem as old as time, you tell someone not to do it and they'll just try harder to do it
